.circleci/config.yml

```diff
@@ -9,149 +9,273 @@ jobs:
-      - run: docker-compose run -e ENV=tests_nds web coverage run ./manage.py test -v 2
-      - run: docker-compose run web coverage xml -o var/coverage.xml
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Prepare environment
+          command: |
+            cp .env-prod.dist .env; cp docker-compose-prod.yml docker-compose.yml
+            docker run -p 5432:5432 --env-file .env -d postgis/postgis:11-2.5
+      - run:
+          name: Load docker image
+          command: |
+            docker load -i /tmp/workspace/20.04.$CIRCLE_WORKFLOW_ID.tar
+            docker tag geotrekce/circleci-admin:$CIRCLE_WORKFLOW_ID geotrekce/admin:latest
+      - run:
+          name: Prepare tests
+          command: docker-compose run web update.sh
+      - run:
+          name: Run tests & export coverage
+          command: |
+            docker-compose run -e ENV=tests_nds web coverage run ./manage.py test -v 2
+            docker-compose run web coverage xml -o var/coverage.xml
       - codecov/upload:
           file: var/coverage.xml
           flags: test_nds
           when: on_success
+    resource_class: large
+
   docker_test_nds_18_04:
     machine:
       image: ubuntu-2004:202201-02
     environment:
       LANG: C.UTF-8
     steps:
       - checkout
-      - run: cp .env-prod.dist .env; cp docker-compose-prod.yml docker-compose.yml
-      - run: docker run -p 5432:5432 --env-file .env -d postgis/postgis:11-2.5
-      - run: docker pull geotrekce/circleci-admin:18.04-$CIRCLE_WORKFLOW_ID
-      - run: docker tag geotrekce/circleci-admin:18.04-$CIRCLE_WORKFLOW_ID geotrekce/admin:latest
-      - run: docker-compose run web update.sh
-      - run: docker-compose run -e ENV=tests_nds web coverage run ./manage.py test -v 2
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Prepare environment
+          command: |
+            cp .env-prod.dist .env; cp docker-compose-prod.yml docker-compose.yml
+            docker run -p 5432:5432 --env-file .env -d postgis/postgis:11-2.5
+      - run:
+          name: Load docker image
+          command: |
+            docker load -i /tmp/workspace/18.04.$CIRCLE_WORKFLOW_ID.tar
+            docker tag geotrekce/circleci-admin:18.04-$CIRCLE_WORKFLOW_ID geotrekce/admin:latest
+      - run:
+          name: Prepare tests
+          command: docker-compose run web update.sh
+      - run:
+          name: Run tests
+          command: |
+            docker-compose run -e ENV=tests_nds web coverage run ./manage.py test -v 2
+
+    resource_class: large
+
   docker_test_js:
-    docker:
-      - image: geotrekce/circleci-admin:$CIRCLE_WORKFLOW_ID
+    machine:
+      image: ubuntu-2004:202201-02
     environment:
       LANG: C.UTF-8
       ENV: tests
       SECRET_KEY: foo
       CUSTOM_SETTINGS_FILE: ""
       CONVERSION_HOST: localhost
       CAPTURE_HOST: localhost
-    working_directory: /opt/geotrek-admin
```

Comment:
Why did it exist before ?

Reply:
before, previous built docker image was used directly by CIrcle. Now, we load image and launch it. 
In bot cases, /opt/geotrek-admin is always default working_directory